### ISSUE TITLE
It seems doesn't work on Windows...

### ISSUE THREAD

**Issue by SmerchProsto** [gpt-omni/mini-omni#28]
My environment:
NVIDIA GeForce GTX 1650
Windows 11
Conda

I did all steps by instruction on git, i installed all dep's by requirements. But it seems doensn't work...

**Comment by mini-omni** [comment]
hi, what the problem? you may give more details, so that someone with WinOS env might help.

**Comment by SmerchProsto** [comment]
Look, I activated an venv by condo, install all deps by requirements file, so I have a problem with Pytorch. I installed it from official site with CUDA ( I tried cuda 11.8 and 12.1) with GPU settings Nvidia
(<code>conda install pytorch==2.3.1 torchvision==0.18.1 torchaudio==2.3.1 pytorch-cuda=12.1 -c pytorch -c nvidia</code>). To consume, after many attempts, I have an error like OSError: [WinError 127]. Error loading "{path}\envs\omni\lib\site-packages\torch\lib\nvfuser_codegen.dll" or one of its dependencies..

**Comment by SmerchProsto** [comment]
It seems like the project can be easily launched on Linux according to the instructions from GIT?

**Comment by kunci115** [comment]
use pip instead of cuda

**Comment by mini-omni** [comment]
> It seems like the project can be easily launched on Linux according to the instructions from GIT?

yes, since I do not have WinOS testing enviroment, I only tested on Linux. Hope other WinOS user might help.

**Comment by 808cn** [comment]
Windows下运行，几乎没问题。

有几点要注意，我遇到的问题如下：
1. 当前python切换为python10， 最新的python11 torch 不一定支持。

2.pypytorch报错了。 torch报错AssertionError: Torch not compiled with CUDA enabled
参考如下，解决：
https://pytorch.org/get-started/previous-versions/

最后用：Wheel -> Linux and Windows  -> CUDA 12.1   的方式解决。
# CUDA 12.1
pip install torch==2.3.1 torchvision==0.18.1 torchaudio==2.3.1 --index-url https://download.pytorch.org/whl/cu121

下载国内速度太慢，下载以后。用下面的命令安装的。
pip install "torch-2.3.1+cu121-cp310-cp310-win_amd64.whl"

3  CUDA_11.8 我安装在D盘了。
把这两个的路径添加到环境变量中。D:\CUDA_11.8\bin 和 D:\CUDA_11.8\lib\x64

4. 如果你没有显卡。代码要稍作修改：
在inference.py及其他一些文件中，将代码中的 'cuda:0' 改为 'cpu'。

4. 进入python环境测试, 运行没有问题的话，torch 基本没问题了。
(cat) C:\Users\asus>python
>>> import torch                                             # 导入torch
>>> print(torch.cuda.is_available())                 #cuda是否可用
>>>print(torch.cuda.get_device_name(0))      #返回设备索引
>>>print(torch.cuda.device_count())              # 返回GPU的数量
>>>print(torch.cuda.current_device())            # 返回当前设备索引
>>> print(torch.rand(3,3).cuda())

**Comment by SmerchProsto** [comment]
謝謝，我會嘗試

**Comment by mini-omni** [comment]
similar issue: https://github.com/gpt-omni/mini-omni/issues/35

**Comment by lowpair** [comment]
希望作者能早日支持中文的语音输出。

**Comment by mini-omni** [comment]
> 希望作者能早日支持中文的语音输出。

hi, 由于暂时没有比较好的中文数据，暂时没有计划出中文版本哈。

I will close it for now, please feel free to re-open.

**Comment by SmerchProsto** [comment]
fd

> **Windows下运行，是可以的，说一下我的做法。**
> 
> 我没有用 conda，但也安装成功了。我用了 venv 的管理办法，具体使用如下： 当前python切换为python10 (作者用的是python10，和作者保持一致即可) -------------------------------------- venv虚拟环境用法 ---------------------------------- 1.创建虚拟环境：在终端中输入以下命令，创建一个名为 venv 的虚拟环境： python -m venv venv

Hello, thank you for the explanation. But, unfortunately, it didn't work. It turned out very unexpectedly, it didn't start, but the error is new, which I haven't encountered. I'll explain my steps:
1. I installed everything as you said, pip install torch ... installed cuda 12.1
2. In environment variables I installed as you said
3. Everything is also in venv
4. I installed the packages that were missing via pip install
5. Python version 3.10

All the same data
My environment:
NVIDIA GeForce GTX 1650
Windows 11

as a result, an error
(venv) ...path\mini-omni> python server.py --ip '0.0.0.0' --port 60808
Traceback (most recent call last):
File "...path\mini-omni\server.py", line 6, in <module>
from inference import OmniInference
File "...path\mini-omni\inference.py", line 24, in <module>
import whisper
File "...path\venv\lib\site-packages\whisper.py", line 69, in <module> libc = ctypes.CDLL(libc_name) File "C:\Users\smerc\AppData\Local\Programs\Python\Python310\lib\ctypes\__init__.py", line 364, in __init__ if '/' in name or '\\' in name: TypeError: argument of type 'NoneType' is not iterable
